### ISSUE TITLE
Train a network for splitting the known classes

### ISSUE THREAD

**Issue by JianXiao-24** [patrickschlachter/osr-using-ics#2]
Hello, thanks your sharing.
Training a network for splitting the known classes by runing function " train_logits_cnn" in "main.py“. However,  ”cnn_latent“ network model is saved for function " train_logits_cnnin" in "cnn_pool.py", which is not trained and " "cnn" network model is trained. But the model "cnn_latent" is used for splitting the known classes. So I dont know the meaning of function "train_logits_cnn".

**Comment by patrickschlachter** [comment]
Hello, thanks for your question.
As the "cnn_latent" model is a subset of the "cnn" model, training the full "cnn" model leads also to training the "cnn_latent" model. The full "cnn" model is trained, because the last softmax layer is required to apply categorical cross-entropy loss. After training, this last layer is thrown away, which is why the "cnn_latent" model is saved and further used.
I hope this helps.
Best regards, Patrick

**Comment by JianXiao-24** [comment]
Hello, thanks your reply.
When the cnn_type==''logits_custom'' , the "latent_layer" and "logits_layer" is same, which is show as
        latent_layer = Reshape(target_shape=(-1,), name='latent_layer')(dense_10)
        logits_layer = Reshape(target_shape=(-1,), name='latent_layer')(dense_10)
However, the "latent_layer" and "logits_layer" is different for the other “cnn_type”.
In my opinion, it will not influence the final result for the whole project and you are right.

**Comment by patrickschlachter** [comment]
Hi, you are right.
The cnn type `logits_cnn` was a special test case in which the two layers are the same. But you can play around with it and change it to `latent_layer = Reshape(target_shape=(-1,), name='latent_layer')(dense_9)`.
Then it is consistent with the other types.
